internal/serve/httphandler/disbursement_handler.go

```diff
@@ -372,6 +368,19 @@ func (d DisbursementHandler) PatchDisbursementStatus(w http.ResponseWriter, r *h
 
 	switch toStatus {
 	case data.StartedDisbursementStatus:
+		var distributionPublicKey string
+		var distributionAccount *schema.DistributionAccount
+		if distributionAccount, err = d.DistributionAccountResolver.DistributionAccountFromContext(ctx); err != nil {
+			httperror.InternalError(ctx, "Cannot get distribution account", err, nil).Render(w)
+			return
+		} else if !distributionAccount.IsStellar() {
+			// TODO: during SDP-1177, refactor StartDisbursement to receive the whole distribution account object, rather than just the public key
+			msg := fmt.Sprintf("expected distribution account to be a STELLAR account but got %q", distributionAccount.Type)
+			httperror.BadRequest(msg, err, nil).Render(w)
+			return
```

Comment:
I didn't include tests for this new error because this call will be refactored and reallocated to a new method in SDP-1177, as mentioned in the TODO.